main.tf

```diff
@@ -160,7 +161,73 @@ data "aws_iam_policy_document" "resource_readonly_access" {
       ]
     }
   }
+}
+
+data "aws_iam_policy_document" "resource_push_access" {
+  count = module.this.enabled ? 1 : 0
+
+  statement {
+    sid    = "PushAccess"
+    effect = "Allow"
+
+    principals {
+      type = "AWS"
+
+      identifiers = var.principals_push_access
+    }
+
+    actions = [
+      "ecr:CompleteLayerUpload",
+      "ecr:GetAuthorizationToken",
+      "ecr:UploadLayerPart",
+      "ecr:InitiateLayerUpload",
+      "ecr:BatchCheckLayerAvailability",
+      "ecr:PutImage",
+    ]
+  }
+
+  dynamic "statement" {
+    for_each = length(var.principals_lambda) > 0 ? [1] : []
+
+    content {
+      sid    = "LambdaECRImageCrossAccountRetrievalPolicy"
+      effect = "Allow"
+      actions = [
+        "ecr:BatchGetImage",
+        "ecr:GetDownloadUrlForLayer"
+      ]
+
+      principals {
+        type        = "Service"
+        identifiers = ["lambda.amazonaws.com"]
+      }
 
```

Comment:
```suggestion
```

Reply:
Ah @kpankonen maybe I get why you're doing this after a 2nd look. Is the idea that we do want to include the lambda permissions in the case that the push access principals is the only set of principals supplied to this module and in that case the lambda principals will still apply?

If so, we'll want to do this another way, but I can deal with that via a separate PR. No need to drag your PR out any longer than it already is.

Reply:
i think i agree it shouldn't be there. i added it based on https://github.com/cloudposse/terraform-aws-ecr/pull/98#issuecomment-1229112488 but i'm happy to remove it

Reply:
It's needed, but I'll do a fast follow on PR to address this and push it forward. The original implementation of `principals_lambda` was flawed, but that isn't your problem.